openhands/server/listen.py

```diff
@@ -754,4 +788,61 @@ async def security_api(request: Request):
     )
 
 
-app.mount('/', StaticFiles(directory='./frontend/dist', html=True), name='dist')
+@app.get('/api/zip-directory')
+async def zip_current_workspace(request: Request):
+    logger.info('Zipping workspace')
+    runtime: Runtime = request.state.session.agent_session.runtime
+    workspace_dir = runtime.config.workspace_mount_path_in_sandbox
+
+    try:
+        zip_file = zip_directory(workspace_dir)
+        response = FileResponse(zip_file, filename="workspace.zip", media_type='application/x-zip-compressed')
+
+        return response
+    except Exception as e:
+        logger.error(f'Error zipping workspace: {e}', exc_info=True)
+        raise HTTPException(
+            status_code=500,
+            detail='Failed to zip workspace',
+        )
```

Comment:
The zip needs to take place inside the runtime, e.g. by sending a `run_command` and then copying the resulting file. That's the only way this will work with a remote runtime